packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/OutputTypes/TypeProvider.cs

```diff
@@ -50,6 +48,8 @@ protected TypeProvider(SourceInputModel? sourceInputModel)
 
         public bool IsStruct => TypeKind is TypeKind.Struct;
 
+        public bool IsValueType => TypeKind is TypeKind.Enum or TypeKind.Struct;
```

Comment:
Should this or IsStruct be on TypeProvider or should you access it through TypeProvider.Type?

Reply:
actually we treat enum and struct differently and they are both value types
We have:
```
bool IsStruct => TypeKind is TypeKind.Struct;
bool IsEnum => TypeKind is TypeKind.Enum;
```
therefore we could change this to `IsStruct || IsEnum` or we remove this and when we need this we just use `IsStruct || IsEnum`
We cannot use `TypeProvider.Type.IsValueType` because the purpose of adding this is to assign the `IsValueType` value with the value here.

Reply:
There are removed per offline discussion